msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/OAuthAuthorizationGrant.java

```diff
@@ -15,74 +13,54 @@
 
 class OAuthAuthorizationGrant extends AbstractMsalAuthorizationGrant {
 
-    private AuthorizationGrant grant;
     private final Map<String, List<String>> params = new LinkedHashMap<>();
 
-    OAuthAuthorizationGrant(final AuthorizationGrant grant, Set<String> scopesSet, ClaimsRequest claims) {
-        this(grant, scopesSet != null ? String.join(" ", scopesSet) : null, claims);
-    }
-
-    String addCommonScopes(String scopes) {
-        Set<String> allScopes = new HashSet<>(
-                Arrays.asList(COMMON_SCOPES_PARAM.split(SCOPES_DELIMITER)));
-
-        if (!StringHelper.isBlank(scopes)) {
-            allScopes.addAll(Arrays.asList(scopes.split(SCOPES_DELIMITER)));
-        }
-        return String.join(SCOPES_DELIMITER, allScopes);
+    OAuthAuthorizationGrant(Map<String, List<String>> params, Set<String> scopesSet, ClaimsRequest claims) {
```

Comment:
issue: I haven't seen a single instance where the param value is a list with more than 1 value. They are all 1 values? This creates unnecessary allocations of List

Reply:
All of them at least have `grant_type`, and some have a few more. For example, `AuthorizationCodeRequest` also has `code`, `redirect_uri`, and `code_verifier`

However, I agree that it's for the most part an unnecessary. I believe the reason for it is because a lot of Nimbus's APIs use `Map<String, List<String>>` for parameters, even though the `List` often has just one element

The next PR will remove the dependency from the classes that handle the HTTP calls where these parameters maps are actually used, and that will include refactoring to simplify these maps